### PR TITLE
bpo-40280: Disable unusable core extension modules on emscripten (GH-29834)

### LINKED ISSUE #10: Disable unusable core extension module
3.11-dev / main has new code in its configure script to disable extension modules for a platform. I propose the disable modules for ``emscripten`` system that don't make sense in a browser environment or are rarely used, e.g. terminal and UI related modules such as curses, termios or tkinter, dbm, multiprocessing, and so on.

* _curses _curses_panel
* _dbm _gdbm
* grp
* _multiprocessing _posixshmem
* nis
* ossaudiodev
* _posixsubprocess
* resource
* _scproxy
* spwd
* syslog
* termios
* _xxsubinterpreters

### PR DESCRIPTION
**Pull request by erlend-aasland**
Based on https://github.com/ethanhs/python-wasm/issues/10#issue-1065408622

Fixes https://github.com/ethanhs/python-wasm/issues/10

Co-authored-by: Christian Heimes <christian@python.org>

https://bugs.python.org/issue40280

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -6191,6 +6191,26 @@ AS_CASE([$ac_sys_system],
   [CYGWIN*], [py_stdlib_not_available="_scproxy nis"],
   [QNX*], [py_stdlib_not_available="_scproxy nis"],
   [FreeBSD*], [py_stdlib_not_available="_scproxy spwd"],
+  [emscripten], [
```
Comment (tiran) [inline configure.ac]: ``ac_sys_system`` for ``*-*-emscripten`` is ``Emscripten``

```suggestion
  [Emscripten], [
```

Comment (erlend-aasland) [reply configure.ac]: Yes, I just noticed that. Fixed in force push :)

**Review by emmatyping — APPROVED** [review]

**Comment by tiran** [comment]
@ethanhs is the list sensible?

**Comment by ethanhs** [comment]
> @ethanhs is the list sensible?

Yes, I think so. It is possible we may add or remove things down the line, but I think this is a good list as far as I am aware of things.

**Pull request merged**